app/components/custom_list/index.js

```diff
@@ -60,10 +60,9 @@ export default class CustomList extends PureComponent {
                 newData = this.props.createSections(data);
             }
 
-            const mergedData = Object.assign({}, newData, this.state.data);
```

Comment:
This only works for a small number of user per letter (e.g. 'A').  In a case where it exceeds base value of 100, the old data (from state.data), overrides the newData.